### PR TITLE
ctpdev: dump raw on error

### PR DESCRIPTION
**Pull request by lietava**

### CODE REVIEW

**Comment by lietava** [comment]
@martenole @shahor02 : I want to dump whole TF  at line 
https://github.com/lietava/AliceO2/blob/7d05b9a39bc4bc723ebb7b4d803f16d0d7a10fcd/Detectors/CTP/reconstruction/src/RawDataDecoder.cxx#L303
Is there some way how to use the DPLRawParser or some other tool for that ?
Cheers, Roman.

**Comment by lietava** [comment]
@shahor02 @martenole : I found way how to dump. Let me know, please, if you have some comments. Otherwise I clean the code and 'ready for review'.

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 1]
```diff
@@ -161,7 +170,7 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
     } else {
       LOG(error) << "Unxpected  CTP CRU link:" << linkCRU;
     }
-    LOG(debug) << "RDH FEEid: " << feeID << " CTP CRU link:" << linkCRU << " Orbit:" << rdhOrbit << " triggerType:" << triggerType;
+    LOG(info) << "RDH FEEid: " << feeID << " CTP CRU link:" << linkCRU << " Orbit:" << rdhOrbit << " triggerType:" << triggerType;
```
Comment (shahor02) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: isn't this too verbose?

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 2]
```diff
@@ -193,6 +202,7 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
     if (mPadding == 1) {
       wordSize = 16;
     }
+    LOG(info) << ii << " payload size:" << payload.size();
```
Comment (shahor02) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: this too?

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 3]
```diff
@@ -286,7 +282,20 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
       digits.push_back(dig.second);
     }
   }
-  return 0;
+  ret = 1;
```
Comment (shahor02) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: I guess this was for the test only?

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 4]
```diff
@@ -286,7 +282,20 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
...
+  if (ret) {
+    if (nwrites < 1) {
+      std::ofstream dumpctp("/tmp/dumpCTP.bin", std::ios::out | std::ios::binary);
```
Comment (shahor02) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: it is worth to check if the file is ok
```
if (!dumpctp.good()) {
  LOGP(error, "Failed to open file {}", "/tmp/dumpCTP.bin");
} else {
  ...
}
```

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx (hunk 1 above)
Comment (lietava) [reply Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: yes

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx (hunk 3 above)
Comment (lietava) [reply Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: yes

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx (hunk 4 above)
Comment (lietava) [reply Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]: yes

**Comment by lietava** [comment]
@shahor02 : thanks, yes, it was only draft. I removed debugs and added option for number of errors to print.

**Comment by davidrohr** [comment]
If the code is ok now, would be great to merge it today. Then we could deploy it tomorrow with the software update.

**Comment by lietava** [comment]
There is one error check but it does not seems to be ctp related.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**